src/python/deepgnn/graph_engine/snark/distributed.py

```diff
@@ -84,3 +92,12 @@ def reset(self):
         """Reset server."""
         if self.server is not None:
             self.server.reset()
+
+    def __reduce__(self):
+        """On serialize reload as Client."""
```

Comment:
why do we reload server as a client? What happens to an old server?

Reply:
Server stays in main thread, when ge is serialized into plasma storage and pulled out in the worker it reloads as client.

Reply:
Does the worker use this client then? If no, then why do we create it. If yes, then a multi-server setup will fail, because client will not be connected to all of them.

Reply:
Yes, it makes the code simpler for the basic user. e.g.

```python
g = Server(address, host)
def transform(batch):
    g # Here this is actually a client
    return model.query(g, batch)
pipeline.map_batches(transform)
```
Otherwise for multi serveres they'll do
```python
s = [Server(address, host), Server(address, host)]
def transform(batch):
    g = DistributedClient([address1, address2])
    return model.query(g, batch)
pipeline.map_batches(transform)
```

Reply:
The transform function should be independent from data setup though.

Reply:
Multiple servers can do this instead to make it independent of setup. I'll add test case for this in dataset.rst
```python
s = [Server(address, host), Server(address, host)]
g = DistributedClient([address1, address2])
def transform(batch):
    return model.query(g, batch)
pipeline.map_batches(transform)
```

Reply:
well, then it means we don't need to create a client on server deserialize at all.